lib/galaxy/jobs/dynamic_tool_destination.py

```diff
@@ -1229,7 +1230,39 @@ def map_tool_to_destination(
     default_priority = 'med'
     priority = default_priority
 
-    if config is not None:
+    # fetch priority information from workflow/job parameters
+    job_parameter_list = job.get_parameters()
+    workflow_params = None
+    job_params = None
+    if job_parameter_list is not None:
+        for param in job_parameter_list:
+            if param.name == "__workflow_resource_params__":
+                workflow_params = param.value
+            if param.name == "__job_resource":
+                job_params = param.value
+
+    # Priority coming from workflow invocation takes precedence over job specific priorities
+    if workflow_params is not None:
+        resource_params = json.loads(workflow_params)
+        if 'priority' in resource_params:
+            # For by_group mapping, this priority has already been validated when the
+            # request was created.
+            if resource_params['priority'] is not None:
+                priority = resource_params['priority']
+
+    elif job_params is not None:
+        resource_params = json.loads(job_params)
+        if 'priority' in resource_params:
+            # Should this really override workflow level priority? -John
+            # TODO: make a decision on what overwrites what. The reasoning behind having workflow > job is that a job
+            # parameter may exist on jobs in a workflow, and if the user runs the workflow with a parameter, they may
+            # not know that the job is overwriting their choice.
```

Comment:
It makes sense for your TDT rules that the workflow priority would take precedents for sure. I just feel like this file treats the parameters without any semantic meaning - they are just keys and values. So it would be better to structure this in such a way that the dynamic tool destinations plugin made that determination. I'll keep looking and see if I see an obvious way to do that.